commit 83d666f858efebbdcc5ffabc16a993cc667f8bd7
Author: Hsiu-Chang Chen <hsiuchangchen@google.com>
Date:   Fri Jul 26 12:47:52 2019 +0800

    Fix build break for Argument list too long
    
    Reduce the number of files listed in the for loop statement
    in function update_lto_symversions to prevent error "Argument
    list too long" when building kernel in local workstation.
    
    Bug: 138277477
    Test: Build pass
    Change-Id: I5218c60e6728f2425f3faa7f795c7782c5f817ad
    Signed-off-by: Hsiu-Chang Chen <hsiuchangchen@google.com>

diff --git a/scripts/Makefile.build b/scripts/Makefile.build
index 87aa4361e172..7a05bafb5d85 100644
--- a/scripts/Makefile.build
+++ b/scripts/Makefile.build
@@ -503,11 +503,10 @@ ifdef CONFIG_LTO_CLANG
     # combine symversions for later processing
     update_lto_symversions =				\
 	rm -f $@.symversions; 				\
-	for i in $(filter-out FORCE,$^); do		\
-		if [ -f $$i.symversions ]; then		\
-			cat $$i.symversions 		\
-				>> $@.symversions;	\
-		fi;					\
+	for i in $(foreach n,						\
+			$(filter-out FORCE,$^),				\
+			$(if $(wildcard $(n).symversions),$(n))); do	\
+		cat $$i.symversions >> $@.symversions;			\
 	done;
   endif
   # rebuild the symbol table with llvm-ar to include IR files
